### ISSUE TITLE
Square Resolutions

### ISSUE THREAD

**Issue by boehmiboy** [raspberrypi/firmware#265]
Is it possible to implement Square resolutions with aspect ratio=1? I have a 640x640 Display connected to the HDMI-Output but the raspberry switches always to 640x480 because it rejects the Edid file. I tried to Set up a custom Resolution but I can't fill in the aspect Ratio Parameter. I can post the Timings of the Display if needed. Tank you very much!

**Comment by Ruffio** [comment]
@boehmiboy is this still an issue?

**Comment by Ruffio** [comment]
@popcornmix please consider closing this issue, due to lack of activity/response to question from 17th of june 2015.

**Comment by JamesH65** [comment]
Closing due to lack of activity. Reopen if you feel this issue is still relevant.

**Comment by steffensq** [comment]
I have the exact same question expect I do not use EDID. 
I am using a 720x720 display and am having trouble getting a proper image.

hdmi_cvt= 720 720 60 does produce an image on the display but shifted entirely. Top left is now cropped and located bottom right, vice versa.

hdmi_timings produces a mess of pixels across a black black ground.

I am assuming this is related to the aspect ratio 1:1. 
Is there a formula to alter the frame buffer resolution with an achievable aspect ratio to produce content on square pixels with a 720x720 resolution?

**Comment by JamesH65** [comment]
You will need to determine the correct HDMI timings for the display (on the datasheet?), set using hdmi_timings, and set up the frame buffer resolutions to match.

**Comment by steffensq** [comment]
I've done that and it doesn't appear to work as expected.
Square resolution in hdmi_timing appear to be more complicated.
With the correct timing based on the spec, HDMI never enables. 
I have a test point available on my conversion board that confirms this.

**Comment by steffensq** [comment]
I stand corrected.
Messing with the hdmi_timing horizontal front porch value did the trick. 
Changing any of the other porch values had no effect on the image.
My spec says min. 19DCLK and max. 103DCLK, 
Setting various values {5,10,15,20,25,40,45,60,65,100} for the front porch all had the same effect, messy pixel data.
Once I set the front porch value to 35, I have a crystal clear image. 
It's frustrating that I had to go through trail and error to figure it out, it was awfully tedious. But I am very glad it is now working and that RPi supports these custom resolutions.
